src/main/java/tdd_module/Solve.java

```diff
@@ -0,0 +1,76 @@
+package tdd_module;
+
+public class Solve {
+
+    public static boolean checkCoef (double[] coefs){
+
+        boolean result = true;
+
+        for (double coef : coefs) {
+            if (Double.isNaN(coef) || Double.isInfinite(coef)) {
+                result = false;
+                return result;
+            }
+        }
+
+        return result;
+
+    }
+
+    public static double[] calcQuadraticEquation(double a, double b, double c) {
+
+        double[] coefArray = new double[] {a,b,c};
+        double[] result = new double[3];
+        double e = 0.0001;
+
+        if ( !checkCoef(coefArray) ) {
+
+            System.out.println("Нельзя задавать не числовые коэфициенты!");
+            return result;
+        }
+
+        if ( Math.abs(a) == 0.00) {
+
+            System.out.println("Коэффициент a не может быть равным 0!");
+            return result;
+        }
+
+        // вычисление дискриминанта
+        double D = (b * b) - (4 * a * c);
+
+        System.out.println("D = " + D);
+
+        if (D > 0.0) {
+            // два различных корня
+            double x1 = (-b + Math.sqrt(D)) / (2 * a);
+            double x2 = (-b - Math.sqrt(D)) / (2 * a);
+
+            System.out.println("x1 = " + x1);
+            System.out.println("x2 = " + x2);
+
+            result[0] = x1;
+            result[1] = x2;
+            result[2] = D;
+
+            return result;
+        } else if (D <= 0.0 + e && D >= 0.0 - e) {
```

Comment:
тоже стоит сравнивать с epsilon:
```
abs(D) < eps
```

Reply:
Так делать нельзя, т.к. в этом случае при Д = 0   попадаем в решение с одним корнем, а это неверно

Reply:
По логике должно быть так?

```suggestion
        } else if (D <  e && D > - e) {
```